src/lex.rs

```diff
@@ -321,8 +324,13 @@ fn is_special(c: u8) -> bool {
             | b'`'
             | b'.'
             | b'$'
-            | b'\n'
-    )
+    );
+    #[cfg(feature = "softbreak")]
+    {
+        // this has a significant impact on performance in paragraphs with linebreaks
+        special |= c == b'\n';
```

Comment:
might be better to try and optimize softbreaks rather than disabling them completely